### PR TITLE
disruptor small improvements

### PR DESCRIPTION
**Pull request by press0**
@jac18281828 
@richkraft 

Hi John & Richard & Walter,  

Great library.  Would like to assist.  Please review 2 small improvements:
* private final method
* method always returns false

Respectfully,

Clay Press
press0@gmail.com

### CODE REVIEW

**Comment by jac18281828** [comment]
Hi Clay, 

I really appreciate the contribution.    

I agree with your change on the final keyword as the "private final" qualifier is a slightly confusing construct.     However, the article you cite is from 2002.     I don't accept "conventional wisdom" when it comes to performance changes nor should you.    Since this modifies the critical section, have you done any benchmarks to show this is a non breaking change?   The target platform for Conversant Disruptor is Intel Haswell/Xeon.  

Not to add to the confusion, but the Java Language Spec also makes the following statement: 

S 8.4.3.3
"At run time, a machine-code generator or optimizer can "inline" the body of a final method, replacing an invocation of the method with the code in its body."

With some benchmarking to prove these methods are still inlined, I believe we can merge this change.   

As far as the second change I do not believe I can accept it as it is a breaking change.   I'll place some comments in the relevant sections.   

One additional note, some of those "simplifications" use Java 8 syntax.   While I support that, it makes it harder for us to support Java 7, required by Log4j2.  Just offering this as food for thought.

File: src/main/java/com/conversantmedia/util/concurrent/DisruptorBlockingQueue.java [hunk 1]
```diff
@@ -443,7 +443,7 @@ public boolean retainAll(Collection<?> c) {
         return new RingIter();
     }
 
-    private final boolean isFull() {
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/concurrent/DisruptorBlockingQueue.java]: This is a critical section and must be inlined to achieve best performance.

File: src/main/java/com/conversantmedia/util/estimation/Percentile.java [hunk 2]
```diff
@@ -173,7 +173,7 @@ public float getMax() {
         return q[2*m+3];
     }
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/estimation/Percentile.java]: I'm not concerned with performance here.   No need to benchmark this change, this is acceptable.

File: src/main/java/com/conversantmedia/util/estimation/PercentileFile.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
  */
 public final class PercentileFile {
 
-    public static final void main(final String[] arg) throws IOException {
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/estimation/PercentileFile.java]: Agree with the change here.   This is also acceptable without a benchmark.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java [hunk 4]
```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
 
     @Override
     public boolean removeAll(Collection<?> c) {
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: This is actually a breaking change.    The method contract is that "true" is returned whenever the class is modified.  I agree its weird but that's from the Java Language, we are not free to alter it here.      Btw, doesn't your implementation always return false?    You might want to revisit that logic as well.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java [hunk 5]
```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
...
-        int numFalses = 0;
-        for (final Object o : c) {
-            if (!remove(o)) numFalses++;
-        }
-        return numFalses > 0;
+         c.forEach(this::remove);
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: This syntax is only available in Java 8.    I wouldn't mind as much if the implementation is correct but its something to consider.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java [hunk 6]
```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
...
-                } else {
+                if (remove(buffer[headSlot])) {
                     // backtrack one step, we just backed values up at this point
                     i--;
                 }
-
             }
         }
 
-        return numFalses > 0;
```
Comment (jac18281828) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: Seems like a breaking change as the test will never succeed.      Did you run the unit tests for this change?  I'm shocked that they didn't fail?     If they didn't please let me know or submit a pr with test coverage for this use case.   We should certainly have coverage for this as it is an FAQ.

**Comment by press0** [comment]
I will split this commit request.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java (hunk 5 above)
Comment (press0) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: Ah. this is removed from the updated commit request.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java (hunk 4 above)
Comment (press0) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: the method contract is not broken because remove(Object) throws an Exception.

File: src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java (hunk 6 above)
Comment (press0) [inline src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java]: the existing unit tests pass.  2 new unit tests added to cover corner cases.
